diff -urN openplacos.orig/INSTALL_GENTOO openplacos/INSTALL_GENTOO
--- INSTALL_GENTOO	1970-01-01 01:00:00.000000000 +0100
+++ INSTALL_GENTOO	2011-01-22 18:12:26.000000000 +0100
@@ -0,0 +1,31 @@
+<<<<<<<< How-to integrate ebuild in overlay tree (version 0.0.1 for example) >>>>>>>>
+
+1. Create a new portage overlay tree :
+	# mkdir -p /usr/local/portage/dev-ruby/openplacos/files
+
+2. Modify /etc/make.conf to integrate new portage tree :
+	# echo "PORTDIR_OVERLAY=/usr/local/portage" >> /etc/make.conf
+
+3. Download latest files on repositary http://pingwho.homelinux.org/pub/gentoo/ftp/overlay
+	# cd /usr/local/portage/dev-ruby/openplacos/files
+	# wget -c http://pingwho.homelinux.org/pub/gentoo/ftp/overlay/dev-ruby/openplacos/files/openplacos-0.0.1-gentoo.diff
+	# cd ..
+	# wget -c http://pingwho.homelinux.org/pub/gentoo/ftp/overlay/dev-ruby/openplacos/openplacos-0.0.1.ebuild
+	# wget -c http://pingwho.homelinux.org/pub/gentoo/ftp/overlay/dev-ruby/openplacos/Changelog
+	
+4. Create checksums Manifest :
+	# ebuild openplacos-0.0.1.ebuild digest
+
+4. Modify package.keywords in portage :
+	#  echo "dev-ruby/openplacos   ~x86" */ Or '~amd64' on 64bits OS platform
+
+5. Now, you can install latest GIT openplacos version with :
+	# USE="gnome" emerge -av openplacos
+
+6. Automatic starting service :
+	# /etc/init.d/openplacos start
+	# rc-update add openplacos default
+
+7. Look at http://openplacos.sourceforge.net/ for more information
+
+<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< End >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
diff -urN openplacos.orig/setup_files/openplacos openplacos/setup_files/openplacos
--- setup_files/openplacos	2011-01-22 18:10:07.000000000 +0100
+++ setup_files/openplacos	2011-01-22 20:11:38.000000000 +0100
@@ -3,6 +3,7 @@
 case "$1" in
     	start)
         	echo "Starting OpenplacOS"
+		export RUBYOPT="rubygems"
 		su -l openplacos -s /bin/sh -c "/usr/bin/openplacos-server /etc/default/openplacos > /tmp/openplacos.log &"
     if [ $? -gt 0 ]; then
 		    exit 1
